src/playlist.rs

```diff
@@ -1048,6 +1081,37 @@ impl ByteRange {
     }
 }
 
+impl Display for ByteRange {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.length)?;
+        if let Some(offset) = self.offset {
+            write!(f, "@{}", offset)?;
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for ByteRange {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<ByteRange, String> {
```

Comment:
Doesn't this do the same as the one in [parser.rs](https://github.com/rutgersc/m3u8-rs/blob/master/src/parser.rs#L735)? Maybe these could share the same implementation?